src/backend/metal/src/device.rs

```diff
@@ -2399,7 +2401,14 @@ impl hal::Device<Backend> for Device {
             levels: 0..raw.mipmap_level_count() as image::Level,
             layers: 0..image.kind.num_layers(),
         };
-        let mtl_type = conv::map_texture_type(kind);
+        let mtl_type = if image.mtl_type == MTLTextureType::D2Multisample {
```

Comment:
There's also `MTLTextureType2DMultisampleArray` in macOS 10.14+, I just opened #2689 to track it